### ISSUE TITLE
Perl Sereal::Encoder use of LIKELY requires Perl 5.8.9

### ISSUE THREAD

**Issue by plicease** [Sereal/Sereal#95]
Trying to build Sereal::Encoder on Perl 5.8.8 and I get these errors in test:

```
dare% prove -bv t/010_desperate.t 
t/010_desperate.t .. /home/ollisg/perl5/perlbrew/perls/perl-5.8.8/bin/perl: symbol lookup error: /home/ollisg/.cpanm/work/1434037476.15559/Sereal-Encoder-3.005/blib/arch/auto/Sereal/Encoder/Encoder.so: undefined symbol: LIKELY
Dubious, test returned 127 (wstat 32512, 0x7f00)
No subtests run 

Test Summary Report
-------------------
t/010_desperate.t (Wstat: 32512 Tests: 0 Failed: 0)
  Non-zero exit status: 127
  Parse errors: No plan found in TAP output
Files=1, Tests=0,  0 wallclock secs ( 0.02 usr  0.01 sys +  0.04 cusr  0.01 csys =  0.08 CPU)
Result: FAIL
```

It seems there is no `LIKELY` (used in `srl_encoder.c`) in Perl 5.8.8.  capntesters show similar errors, though with different symbols.

**Comment by ikruglov** [comment]
Hi. Thank you for your report. The issues is already fixed and will be included in next release.

**Comment by plicease** [comment]
Appreciate your speedy response!  Unfortunately now it is complaining on `DPPP_my_newSV_type`:

```
dare% prove -bv t/400_evil.t 
t/400_evil.t .. 
ok 1
ok 2 - serializing tied hash did not die
ok 3 - serializing tied hash returns string
ok 4 - deserializing tied hash did not die
ok 5 - deserializing tied hash yields defined output
ok 6 - deserializing tied hash yields expected output
ok 7
ok 8 - serializing tied array did not die
ok 9 - serializing tied array returns string
ok 10 - deserializing tied array did not die
ok 11 - deserializing tied array yields defined output
ok 12 - deserializing tied array yields expected output
ok 13
ok 14 - serializing tied scalar did not die
ok 15 - serializing tied scalar returns string
ok 16 - deserializing tied scalar did not die
ok 17 - deserializing tied scalar yields defined output
ok 18 - deserializing tied scalar yields expected output
ok 19 - encode does not segfault
ok 20 - __DIE__ called, encode 2 did not die (2)
/home/ollisg/perl5/perlbrew/perls/perl-5.8.8/bin/perl: symbol lookup error: /home/ollisg/dev/Sereal/Perl/Encoder/blib/arch/auto/Sereal/Encoder/Encoder.so: undefined symbol: DPPP_my_newSV_type
Dubious, test returned 127 (wstat 32512, 0x7f00)
All 20 subtests passed 

Test Summary Report
-------------------
t/400_evil.t (Wstat: 32512 Tests: 20 Failed: 0)
  Non-zero exit status: 127
  Parse errors: No plan found in TAP output
Files=1, Tests=20,  0 wallclock secs ( 0.02 usr  0.01 sys +  0.05 cusr  0.02 csys =  0.10 CPU)
Result: FAIL
```

**Comment by ikruglov** [comment]
argh... okay, I'll try to build Sereal on 5.8.8 later tonight.

**Comment by andyjones** [comment]
The tests pass on 5.8.8 for me by using the global variant of `newSV_type` or requesting it in srl_encoder.c: https://gist.github.com/andyjones/d37f20e5c40599b51d6b

Also checked 5.16.3 and they continue to pass with either patch

**Comment by demerphq** [comment]
Has this been properly resolved?

On 20 September 2015 at 15:49, Andy Jones <notifications@github.com> wrote:

> The tests pass on 5.8.8 for me by using the global variant of newSV_type
> or requesting it in srl_encoder.c:
> https://gist.github.com/andyjones/d37f20e5c40599b51d6b
>
> Also checked 5.16.3 and they continue to pass with either patch
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Sereal/Sereal/issues/95#issuecomment-141786206>.
>



-- 
perl -Mre=debug -e "/just|another|perl|hacker/"

**Comment by andyjones** [comment]
pull request #108 worked for me on 5.8.8

I also had to change the Makefile.PL as my old version of GCC (4.1.2) doesn't support the `-Werror=declaration-after-statement` flag that was added after the latest CPAN release. I'll raise a separate issue for that

**Comment by demerphq** [comment]
I believe that I have properly merged your fixes. Thanks. 

They should be available in 3.006_002 and later in 3.007
